store/localcachelayer/channel_layer.go

```diff
@@ -132,3 +152,50 @@ func (s LocalCacheChannelStore) GetPinnedPostCount(channelId string, allowFromCa
 
 	return count, nil
 }
+
+func (s LocalCacheChannelStore) SaveMember(member *model.ChannelMember) (*model.ChannelMember, *model.AppError) {
+	defer s.InvalidateAllChannelMembersForUser(member.UserId)
+	return s.ChannelStore.SaveMember(member)
+}
+
+func (s LocalCacheChannelStore) IsUserInChannelUseCache(userId string, channelId string) bool {
+	ids, err := s.GetAllChannelMembersForUser(userId, true, false)
+	if err != nil {
+		mlog.Error("Error getting all channel members for user", mlog.Err(err))
+		return false
+	}
+
+	if _, ok := ids[channelId]; ok {
+		return true
+	}
+
+	return false
+}
+
+func (s LocalCacheChannelStore) GetAllChannelMembersForUser(userId string, allowFromCache bool, includeDeleted bool) (map[string]string, *model.AppError) {
+	cache_key := userId
+	if includeDeleted {
+		cache_key += "_deleted"
+	}
+	if allowFromCache {
+		if cacheItem := s.rootStore.doStandardReadCache(s.rootStore.allChannelMembersForUserCache, cache_key); cacheItem != nil {
+			if s.rootStore.metrics != nil {
+				s.rootStore.metrics.IncrementMemCacheHitCounter("All Channel Members for User")
+			}
+			ids := cacheItem.(map[string]string)
+			return ids, nil
+		}
+	}
+
+	if s.rootStore.metrics != nil {
```

Comment:
Same here the `doStandardReadCache` is already generating the increment in metrics